python/huggingfaceserver/huggingfaceserver/__main__.py

```diff
@@ -126,6 +120,14 @@ def list_of_strings(arg):
         choices=dtype_choices,
         help=f"data type to load the weights in. One of {dtype_choices}. Defaults to float16 for GPU and float32 for CPU systems",
     )
+    # vLLM uses max-model-len as paramter to denote max tokens. Register the same for HuggingFace (if vllm not available)
+    parser.add_argument(
+        "--max-model-len",
```

Comment:
All the other args are using `_`(underscore). We should be consistent with the naming.

Reply:
vLLM uses `-` hyphenated separation. Using the same to ensure we can make use of only single parameter name.

Reply:
Wouldn't be a breaking change for current users ?

Reply:
Yeah this would be a breaking change for existing users.
I'm curious if they're internally handling the setting of parameters with diff name for diff backends.

Reply:
@yuzisun What's your thought on this ?

Reply:
if there any strong reason to take the naming convention from vllm ?

Reply:
Someone using kserve need not switch between 2 parameter names if they are switching between HF and vLLM backends. Ensures consistency.

Reply:
We can potential support both conventions if there is a way we can easily convert between the two.

Reply:
If we want to support both, one of them should take precedence right?
As for converting, it is a 1-1 swap. For users, it'd be easier to manage one parameter ig.

Reply:
this should be possible:

```python
arg_parser.add_argument('--example-one', '--example_one')
```

Reply:
So I just added `max_length` back which internally sets `args.max_model_len`. 
Let me know if this works

Reply:
@yuzisun @sivanantha321 @spolti take a look

Reply:
Can we support both `args.max_model_len` and `args.max-model-len` and deprecate `max_length` ?